### PR TITLE
Pulse page from Insights Tab

### PR DESCRIPTION
**Pull request by andreydro**

### CODE REVIEW

File: .idea/.generators [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (kavalenka) [inline .idea/.generators]: Please add .idea/ folder to gitignore. These files generated by your IDE.

File: app/assets/javascripts/insights.coffee [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (kavalenka) [inline app/assets/javascripts/insights.coffee]: Do we really need this empty file in commit?

File: app/assets/stylesheets/insights.scss [hunk 3]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Insights controller here.
```
Comment (kavalenka) [inline app/assets/stylesheets/insights.scss]: One more empty file.

File: app/views/insights/_code_frequency.html.erb [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<!DOCTYPE html>
```
Comment (kavalenka) [inline app/views/insights/_code_frequency.html.erb]: What does the first row do? Do we need it?

File: app/views/insights/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,39 @@
+<!DOCTYPE html>
+<html>
+  <head>
```
Comment (kavalenka) [inline app/views/insights/show.html.erb]: See comment above regarding <head> tag.

File: app/views/insights/_contributors.html.erb [hunk 6]
```diff
@@ -0,0 +1,65 @@
+<!DOCTYPE html>
+<html>
+<head>
```
Comment (kavalenka) [inline app/views/insights/_contributors.html.erb]: I think this part of code should be in application.html.erb.

File: config/routes.rb [hunk 7]
```diff
@@ -1,3 +1,11 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  get '/show', to: 'insights#show'
+  get '/contributors', to: 'insights#_contributors'
```
Comment (kavalenka) [inline config/routes.rb]: InsightsController has only one method #show. Does these routes work? 
Also, where did you get this naming starting from '_' ?

**Review by kavalenka — CHANGES_REQUESTED** [review]

File: config/routes.rb (hunk 7 above)
Comment (andreydro) [reply config/routes.rb]: Working well! It give you a chance to name your route in a way you want. Got it here: https://www.railstutorial.org/book/filling_in_the_layout

File: app/views/insights/show.html.erb (hunk 5 above)
Comment (andreydro) [reply app/views/insights/show.html.erb]: solved

File: app/views/insights/_contributors.html.erb (hunk 6 above)
Comment (andreydro) [reply app/views/insights/_contributors.html.erb]: solved

File: app/views/insights/_code_frequency.html.erb (hunk 4 above)
Comment (andreydro) [reply app/views/insights/_code_frequency.html.erb]: solved

File: .idea/.generators (hunk 1 above)
Comment (andreydro) [reply .idea/.generators]: solved

File: app/assets/javascripts/insights.coffee (hunk 2 above)
Comment (andreydro) [reply app/assets/javascripts/insights.coffee]: solved

File: app/assets/stylesheets/insights.scss (hunk 3 above)
Comment (andreydro) [reply app/assets/stylesheets/insights.scss]: solved

File: .idea/.generators (hunk 1 above)
Comment (kavalenka) [reply .idea/.generators]: So, push your changes to proceed pull request review.

File: app/views/insights/_contributors.html.erb [hunk 8]
```diff
@@ -0,0 +1,105 @@
+<html>
+  <head>
```
Comment (kavalenka) [inline app/views/insights/_contributors.html.erb]: ``<html>``, ``<head>``, ``<body>`` tags are already in application.html.erb
I do not think they should be here.

Comment (kavalenka) [reply app/views/insights/_contributors.html.erb]: Also file names starting with underscore are reserved for partials.
Please have a look at http://guides.rubyonrails.org/layouts_and_rendering.html#using-partials

File: app/views/insights/show.html.erb [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
```
Comment (kavalenka) [inline app/views/insights/show.html.erb]: One more place with tags.

**Review by kavalenka — CHANGES_REQUESTED** [review]

File: app/views/insights/contributors.html.erb [hunk 10]
```diff
@@ -0,0 +1,88 @@
+<div class="container new-discussion-timeline experiment-repo-nav">
+  <div class="repository-content">
+    <div class="columns">
+      <div class="column one-fourth">
+        <nav class="menu" data-pjax>
+          <a href="/pulse" class=" js-selected-navigation-item selected menu-item" role="tab">Pulse</a>
```
Comment (kavalenka) [inline app/views/insights/contributors.html.erb]: Could we use rails method link_to here?

File: app/views/insights/pulse.html.erb [hunk 11]
```diff
@@ -0,0 +1,222 @@
+<div class="container new-discussion-timeline experiment-repo-nav">
+  <div class="repository-content">
+    <div class="columns">
+      <div class="column one-fourth">
+        <nav class="menu" data-pjax>
+          <a href="/pulse" class=" js-selected-navigation-item selected menu-item" role="tab">Pulse</a>
```
Comment (kavalenka) [inline app/views/insights/pulse.html.erb]: Same link_to here.

**Review by kavalenka — CHANGES_REQUESTED** [review]

File: app/controllers/insights_controller.rb [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+
+  def dependency_graph
+  end
+
+  def network
+  end
+
+  def forks
+  end
+
```
Comment (kavalenka) [inline app/controllers/insights_controller.rb]: Please delete empty line.

File: app/views/insights/community.html.erb [hunk 13]
```diff
@@ -0,0 +1,114 @@
+<div class="container new-discussion-timeline experiment-repo-nav">
+  <div class="repository-content">
+    <div class="columns">
+      <div class="column one-fourth">
+        <nav class="menu" data-pjax>
```
Comment (kavalenka) [inline app/views/insights/community.html.erb]: I see duplication of ``<nav>`` tag in some files. What about creation of partial named ```_menu.html.erb```?

**Review by kavalenka — CHANGES_REQUESTED** [review]

File: app/views/insights/_menu.html.erb [hunk 14]
```diff
@@ -0,0 +1,11 @@
+<nav class="menu" data-pjax>
```
Comment (kavalenka) [inline app/views/insights/_menu.html.erb:1]: Nice!

File: app/views/insights/community.html.erb [hunk 15]
```diff
@@ -0,0 +1,104 @@
+<div class="container new-discussion-timeline experiment-repo-nav">
+  <div class="repository-content">
+    <div class="columns">
+      <div class="column one-fourth">
+        <%= render 'insights/menu' %>
```
Comment (kavalenka) [inline app/views/insights/community.html.erb]: Here we should use render_partial method. Please search for the difference between render and render_partial methods.

File: app/views/insights/contributors.html.erb [hunk 16]
```diff
@@ -0,0 +1,78 @@
+<div class="container new-discussion-timeline experiment-repo-nav">
+  <div class="repository-content">
+    <div class="columns">
+      <div class="column one-fourth">
+        <%= render 'insights/menu' %>
```
Comment (kavalenka) [inline app/views/insights/contributors.html.erb]: See comment above.

File: app/views/insights/dependency_graph.erb [hunk 17]
```diff
@@ -0,0 +1 @@
+<h1>Dependency graph</h1>
```
Comment (kavalenka) [inline app/views/insights/dependency_graph.erb]: Shouldn't this file have .html.erb in the end?

**Review by kavalenka — CHANGES_REQUESTED** [review]

File: app/views/insights/community.html.erb (hunk 15 above)
Comment (andreydro) [reply app/views/insights/community.html.erb]: As I get "render" used for one time view. And "render partial" is reusable in a few views

**Comment by kavalenka** [comment]
And the last thing. Please squash all your commits in one commit.
Google for ```git rebase -i HEAD```

File: app/views/insights/forks.erb [hunk 18]
```diff
@@ -0,0 +1 @@
+<h1>Forks</h1>
```
Comment (kavalenka) [inline app/views/insights/forks.erb]: Please add .html.erb in the end of file name.

File: app/views/insights/network.erb [hunk 19]
```diff
@@ -0,0 +1 @@
+<h1>Network</h1>
```
Comment (kavalenka) [inline app/views/insights/network.erb]: Same here.

File: test/controllers/insights_controller_test.rb [hunk 20]
```diff
@@ -0,0 +1,9 @@
+require 'test_helper'
+
+class InsightsControllerTest < ActionDispatch::IntegrationTest
+  test "should get show" do
+    get insights_show_url
+    assert_response :success
+  end
+
```
Comment (kavalenka) [inline test/controllers/insights_controller_test.rb]: Please delete empty line.

File: app/helpers/insights_helper.rb [hunk 21]
```diff
@@ -0,0 +1,2 @@
+module InsightsHelper
```
Comment (kavalenka) [inline app/helpers/insights_helper.rb]: Do we need this empty file?

**Review by kavalenka — APPROVED** [review]

**Pull request merged**